File: front/src/services/user/user.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';
import { User, UserCreateRequest, UserUpdateRequest } from 'src/models/user.model';
import { AuthService } from '../auth/auth.service';

@Injectable({
  providedIn: 'root'
})
export class UserService {
  private apiUrl = 'http://localhost:8080/api/users';
  constructor(
    private http: HttpClient,
    private authService: AuthService
  ) {}

  private getHeaders(): HttpHeaders {
    const token = this.authService.getToken();
    return new HttpHeaders({
      'Content-Type': 'application/json'
    });
  }

  getAllUsers(): Observable<User[]> {
    return this.http.get<User[]>(this.apiUrl, { headers: this.getHeaders() });
  }

  getUserById(id: number): Observable<User> {
    return this.http.get<User>(`${this.apiUrl}/${id}`, { headers: this.getHeaders() });
  }

  createUser(user: UserCreateRequest): Observable<User> {
    return this.http.post<User>(this.apiUrl, user, { headers: this.getHeaders() });
  }

  updateUser(id: number, user: UserUpdateRequest): Observable<User> {
    return this.http.put<User>(`${this.apiUrl}/${id}`, user, { headers: this.getHeaders() });
  }

  deleteUser(id: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}/${id}`, { headers: this.getHeaders() });
  }
}

File: back/src/main/java/com/raf/raf_cloud_back/service/UserService.java
package com.raf.raf_cloud_back.service;

import com.raf.raf_cloud_back.dto.*;
import com.raf.raf_cloud_back.model.User;
import org.apache.coyote.BadRequestException;

import java.util.List;

public interface UserService {

    public LoginResponse login(LoginRequest loginRequest);

    public List<UserResponse> getAllUsers();

    public UserResponse getUserById(Long id);

    public UserResponse createUser(UserCreateRequest request) ;

    public UserResponse updateUser(Long id, UserUpdateRequest request);

    public void deleteUser(Long id);


}
